bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/graphics/Image.java

```diff
@@ -778,6 +778,7 @@ public Image(Device device, ImageFileNameProvider imageFileNameProvider) {
 	super(device);
 	if (imageFileNameProvider == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
 	this.imageFileNameProvider = imageFileNameProvider;
+	//TODO: implement fine-grained zoom handling here as well?
```

Comment:
For Mac the implementation seems to be completely different than for the other platforms and it even looks to me that anything else than 100% and 200% is not considered?

Reply:
From my understanding, MacOS/Cocoa is only capable of 100% and 200% displays and thus everything in the SWT implementation for MacOS is to 100/200 as well.

Reply:
I have now implemented something that seemed suitable just from looking at the code on a Windows computer (no way to try it out), but I'm absolutly not sure if that's reasonable. And I have the impression that code should also be enhanced, because currently it loads the image twice in some cases which might be unnecessary and could be avoided.

In any case it would be good to get support from somebody who has a Mac at hand to try this out and to get some details.

Reply:
Unfortunately, MacOS has no on-demand loading for image data at different zooms implemented yet. Thus, whenever possible, two representations of each image are created upon image instantiation. We need to implement this here as well, which is why the changes are generally fine.
I still have enhanced the changes with support for the constructors taking `InputStream` and `String` to also render multiple sizes of SVGs in case they refer to an SVG. In addition, I have revised the constructor accepting an `ImageFileNameProvider` to not unnecessary try to rescale the 200% image but just add a representation for it as is, like done in other constructors as well.